packages/tentales/src/types/index.d.ts

```diff
@@ -0,0 +1,72 @@
+import * as Koa from "koa"
+import { Log } from "tentales-log"
+
+export interface Config {
+  port: number
+  reactComponentsDirectory: string
+  services: {
+    renderer: ServiceConfig
+    data: ServiceConfig
+    editor: ServiceConfig
+  }
+  hooks?: {
+    middlewares?: {
+      first?: Middleware[]
+      beforeRender?: Middleware[]
+      render?: Middleware[]
+      afterRender?: Middleware[]
+      beforeRendererService?: Middleware[]
+      rendererService?: Middleware[]
+      afterRendererService?: Middleware[]
+      beforeEditorService?: Middleware[]
+      editorService?: Middleware[]
+      afterEditorService?: Middleware[]
+      beforeDataService?: Middleware[]
+      dataService?: Middleware[]
+      afterDataService?: Middleware[]
+      last?: Middleware[]
+    }
+  }
+}
+
+export interface Action {
```

Comment:
Maybe we could make this a generic?

```ts
Action<T> {
  // ...
  {
    [property: string]: T
  }
}
```

Reply:
Check out 23c7828, does that make sense?

Reply:
Yes :D nice!

Reply:
🎉